libamqpprox/amqpprox_server.cpp

```diff
@@ -194,18 +196,22 @@ void Server::doAccept(int port, bool secure)
         return;
     }
 
-    d_socket.refreshSocketContext();
+    std::shared_ptr<MaybeSecureSocketAdaptor> incomingSocket =
+        std::make_shared<MaybeSecureSocketAdaptor>(
+            d_ioService, d_ingressTlsContext, false);
```

Comment:
The only thing I'd be worried about is reusing the context between all the sockets and modifying it after the fact which we probably shouldn't support? once the listener is open. 
Potentially a lifetime issue also if the context doesn't outlast the sessions? 

No doubt though this is an improvement.

Reply:
Good question. I checked, and found that the ssl context is held by the Server object, which lives longer than the asio event loop. It also turned out the ssl context wasn't needed after the object was created, so I stopped holding it.

There were a few comments which looked pretty hard-earned in the code I've deleted so could be worth running past @alaric . I think the change is safe because we don't use a MaybeSecureSocketAdaptor object after it's been moved, and by creating the socket where we used to call `refreshSocketContext` we don't need that anymore. Valgrind is happy with this change at least.

Reply:
Yeah I think this is fine since the construction happens just before the async_accept and it will last into the callback. Though I think the setSecure call may be able to be removed here because you have this info when constructing and you might as well set it then?